pkg/collector/corechecks/cluster/kubernetes_state_transformers.go

```diff
@@ -42,6 +46,84 @@ var (
 	}
 )
 
+// now allows testing
+var now = time.Now
+
+// cronJobNextScheduleTransformer sends a service check to alert if the cronjob's next schedule is in the past
+func cronJobNextScheduleTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {
+	message := ""
+	var status metrics.ServiceCheckStatus
+	timeDiff := int64(metric.Val) - now().Unix()
+	if timeDiff >= 0 {
+		status = metrics.ServiceCheckOK
+	} else {
+		status = metrics.ServiceCheckCritical
+		message = fmt.Sprintf("The cron job check scheduled at %s is %d seconds late", time.Unix(int64(metric.Val), 0).UTC(), -timeDiff)
+	}
+	s.ServiceCheck(ksmMetricPrefix+"cronjob.on_schedule_check", status, "", tags, message)
+}
+
+// jobCompleteTransformer sends a service check based on kube_job_complete
+func jobCompleteTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {
+	jobServiceCheck(s, metric, metrics.ServiceCheckOK, tags)
+}
+
+// jobFailedTransformer sends a service check based on kube_job_failed
+func jobFailedTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {
+	jobServiceCheck(s, metric, metrics.ServiceCheckCritical, tags)
+}
+
+var jobTagPattern = regexp.MustCompile(`(-\d{4,10}$)`)
```

Comment:
nit: not sure if `jobTag` is the most descriptive name for that + an example would not hurt

Reply:
renamed it `jobTimestampPattern` and added few comments about it